lib/screens/admin_new_product_page.dart

```diff
@@ -218,15 +222,30 @@ class _AdminNewProductPageState extends State<AdminNewProductPage> {
 
   @override
   Widget build(BuildContext context) {
-    return WillPopScope(
-      onWillPop: _displayConfirmExitDialog,
-      child: Scaffold(
-        appBar: _buildAppBar(),
-        // Build a form to input new product details
-        body: _buildNewProductForm(context),
-        key: _scaffoldKey,
-      ),
-    );
+    return StreamBuilder<QuerySnapshot>(
```

Comment:
I see that you are building the _AdminNewProductPageState widget using the StreamBuilder tied to the colors db stream. My question is: how will you modify this when we have to support dynamic sizes (similar to colors) in the next release?

Reply:
@dragonfly91 Thanks for you comment! These 4 screens and widgets are leaf components that use color related method. As we discussed earlier in Slack, I think this is a way we decided to go about.
- lib/screens/admin_new_product_page.dart
- lib/screens/admin_product_detail_page.dart
- lib/screens/customer_product_detail_page.dart
- lib/widgets/stock_selector.dart

---
I briefly checked a syntax to call 2 collections to get `color` and possibly `size` later. These are 2 options I've just found, not 100% sure if it works though.

1. Combination of StreamBuilder and FutureBuilder
https://stackoverflow.com/questions/57204174/flutter-firestore-query-2-collections
2. Collection group queries
https://firebase.google.com/docs/firestore/query-data/queries#collection-group-query